torch/distributed/fsdp/_wrap_utils.py

```diff
@@ -62,6 +63,13 @@ def _auto_wrap(
                 root_module, mixed_precision._module_classes_to_ignore
             )
             _warn_on_overridden_mixed_precision(overridden_module_classes)
+        use_orig_params = fsdp_kwargs.get("use_orig_params", False)
```

Comment:
You don't have to fix this now, but it seems weird to only consult the `use_orig_params` on the root `fsdp_kwargs`, when technically it could vary from module to module in `target_module_to_kwargs`. Some extra generality that doesn't have a payoff yet, it seems like.

Reply:
Actually, it doesn't even seem that hard to use the correct per-module kwargs for use_orig_params as you traverse module to module.

Reply:
We expect `use_orig_params` to be uniform across different FSDP instances, and we check for this at lazy initialization time (when we enforce that FSDP wrapping has been finalized).
https://github.com/pytorch/pytorch/blob/aaaafa1bcfd5f89eb57636f423b66bb236ebdd97/torch/distributed/fsdp/_runtime_utils.py#L277-L281
We can and possibly should move these checks up to construction time for auto wrapping (for which the FSDP wrapping is meant to be finalized up front).

Reply:
Once you have LambdaWrapPolicy there is no longer a constraint that it's uniform though, IIUC?